pkg/services/alerting/conditions/query.go

```diff
@@ -333,11 +333,13 @@ func FrameToSeriesSlice(frame *data.Frame) (tsdb.TimeSeriesSlice, error) {
 		switch {
 		case field.Config != nil && field.Config.DisplayName != "":
 			ts.Name = field.Config.DisplayName
-		case field.Labels != nil:
+		case field.Labels != nil && len(field.Labels) > 0:
```

Comment:
You can get the length of nil slices in Go, so the idiom is just the following:
```suggestion
		case len(field.Labels) > 0:
```

Reply:
Nice tip 🙆‍♂️